src/api.jl

```diff
@@ -12,114 +12,259 @@ import .OptimizeElbo
-  trimmed_tiled_blob = ModelInit.trim_source_tiles(active_s, mp, tiled_blob;
-                                                   noise_fraction=0.1);
-  trimmed_tiled_blob, mp, active_s, s
+    return out
 end
 
 
 """
-Fit the Celeste model to a set of sources and write the output to a JLD file.
+infer(ra_range, dec_range, fieldids, frame_dirs; ...)
+
+Fit the Celeste model to sources in a given ra, dec range.
 
-Args:
-  - dir: The directory containing the FITS files.
-  - run: An ASCIIString with the six-digit run number, e.g. "003900"
-  - camcol: An ASCIIString with the camcol, e.g. "6"
-  - field: An ASCIIString with the four-digit field, e.g. "0269"
-  - outdir: The directory to write the output jld file.
-  - partnum: Which of the 1:parts catalog entries to fit.
-  - parts: How many parts to divide the catalog entries into
+- ra_range: minimum and maximum RA of sources to consider.
+- dec_range: minimum and maximum Dec of sources to consider.
+- fieldids: Array of run, camcol, field triplets that the source occurs in.
+- frame_dirs: Directories in which to find each field's frame FITS files.
 
 Returns:
-  Writes a jld file to outdir containing the optimization output.
+
+- Dictionary of results, keyed by SDSS thing_id.
 """
-function infer(
-      dir::AbstractString, run::AbstractString, camcol::AbstractString,
-      field::AbstractString, outdir::AbstractString,
-      partnum::Int, parts::Int)
-
-    # get images
-    images = SkyImages.load_sdss_blob(dir, run, camcol, field)
-
-    # load catalog and convert to Array of `CatalogEntry`s.
-    cat_df = SDSS.load_catalog_df(dir, run, camcol, field)
-    cat_entries_all = SkyImages.convert_catalog_to_celeste(cat_df, images)
-
-    # Filter low-flux objects.
-    flux_cols = [ symbol("psfflux_$b") for b in band_letters ]
-    max_fluxes =
-      Float64[ maximum([ cat_df[row, flux_col] for flux_col in flux_cols ])
-               for row in 1:size(cat_df, 1) ];
-    good_rows = max_fluxes .> 2;
-    println("Keeping $(sum(good_rows) / length(good_rows)) of the objects.")
-    cat_entries = cat_entries_all[good_rows]
-
-    # limit to just the part of the catalog specified.
-    partsize = length(cat_entries) / parts
-    minidx = round(Int, partsize * (partnum - 1)) + 1
-    maxidx = round(Int, partsize * partnum)
-
-    objids = [ entry.objid for entry in cat_entries[minidx:maxidx] ]
+function infer(ra_range::Tuple{Float64, Float64},
+               dec_range::Tuple{Float64, Float64},
+               fieldids::Vector{Tuple{Int, Int, Int}},
+               frame_dirs::Vector;
+               fpm_dirs=frame_dirs,
+               psfield_dirs=frame_dirs,
+               photofield_dirs=frame_dirs,
+               photoobj_dirs=frame_dirs)
+
+    # the code below assumes there is at least one field.
+    if length(fieldids) == 0
+        return Dict{Int, Dict}()
+    end
+
+    # Read in all photoobj catalogs.
+    rawcatalogs = Array(Dict, length(fieldids))
```

Comment:
All this code looks like something you might want to run outside infer (e.g. in interactive mode for exploration or debugging).  Could you wrap it in a function rather than including it in infer()?